### PR TITLE
Narwallets

### PR DESCRIPTION
**Pull request by Calcifer1001**
Thanks for taking the time to submit a pull request!

If you haven't already, be sure to take a look at the [CONTRIBUTING](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md) documentation to get up to speed with the practices in this repository.

When you're ready, switch to the "Preview" tab to select an applicable template:

- [General template (most common)](?expand=1&template=general.md)
- [Wallet integration template](?expand=1&template=wallet_integration.md)

### CODE REVIEW

**Comment by kujtimprenkuSQA** [comment]
Hey, @Calcifer1001 thank you for submitting this PR.

When you create a PR before submitting the PR you should go to the "Preview" tab of the description of the PR and choose the **Wallet integration template** option, you might need to close the current PR and open a new one and choose that option because otherwise the GH-Actions will not be triggered.

I am unable to build the package because there are a few errors, can you please remove `node_module`s and `dist` folder then try to install all dependencies and make a fresh build?

File: packages/narwallets/src/lib/injected-narwallets.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
+// Empty string if we haven't signed in before.
```
Comment (kujtimprenkuSQA) [inline packages/narwallets/src/lib/injected-narwallets.ts:1]: Is this file relevant, are you making use of these types are these the correct types that your wallet uses they seem to be a copy of the Sender wallet types?

File: package.json [hunk 2]
```diff
@@ -141,18 +142,18 @@
...
     "nx": "14.4.2",
-    "gh-pages": "^4.0.0",
     "prettier": "^2.7.1",
     "react-test-renderer": "18.2.0",
     "sass": "1.52.3",
     "semantic-release": "^19.0.2",
     "ts-jest": "27.1.4",
     "ts-node": "~10.8.0",
-    "typescript": "~4.7.2",
+    "typescript": "^4.8.4",
```
Comment (kujtimprenkuSQA) [inline package.json:156]: Why did you update the Typescript version here?

File: packages/narwallets/src/lib/narwallets.ts [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+  params?: any
+): Promise<any> => {
+  const promise = new Promise<any>((resolve, reject) => {
+    id++;
+    let promiseTimeout;
+    if (withTimeout) {
+      promiseTimeout = setTimeout(() => {
+        return reject(Error("timeout"));
+      }, 1000);
+    }
```
Comment (kujtimprenkuSQA) [inline packages/narwallets/src/lib/narwallets.ts:95]: This is always throwing an error and the Narwallets option is not showing up in the modal so I can not test the functionality of this wallet.
Error in cosole: **Error: timeout at eval (narwallets.ts?4493:93:23)**

**Review by kujtimprenkuSQA — COMMENTED** [review]
There are some things that need to be cleaned up and improved please see the comments below.

File: packages/narwallets/src/lib/narwallets.ts [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+        actions: transformActions(transaction.actions),
+      };
+    });
+  };
+
+  if (await isSignedIn()) {
+    setupEvents();
+  }
+
+  return {
```
Comment (kujtimprenkuSQA) [inline packages/narwallets/src/lib/narwallets.ts:269]: Can you please add a new method called `verifyOwner` in the return here?
If your wallet does not support it you can throw an error as it's done here:
https://github.com/near/wallet-selector/blob/main/packages/ledger/src/lib/ledger.ts#L223-L227

**Comment by AmmarHumackicSQA** [comment]
Hi @Calcifer1001 when can we expect these comments to be addressed? Thanks!

**Comment by kujtimprenkuSQA** [comment]
Hey, @Calcifer1001 thank you for the update.

Yes, whenever you're ready please open a new PR and follow the instructions in the description of the PR when you do because in this PR the gh-actions can not be triggered.
After you submit the new PR we will review it again.

**Comment by Calcifer1001** [comment]
For this integration, we had to made some changes to our chrome extension, and we are uploading a new version. It is currently finished, awaiting for google's review. Is it ok to create the PR anyways? Currently, the wallet-selector code, for installing, is pointing to the version that is about to get deprecated. 

Thank you

**Comment by kujtimprenkuSQA** [comment]
Feel free to open the new PR whenever you're ready.

**Comment by kujtimprenkuSQA** [comment]
Closing this PR since the new one has been submitted #503

**Pull request closed without merging**